### PR TITLE
Migrated winreg AppCompatCache construct-based plugin to use dtfabric #1893

### PR DESCRIPTION
**Pull request by joachimmetz**
Migrated winreg AppCompatCache construct-based plugin to use dtfabric #1893

### CODE REVIEW

File: plaso/parsers/winreg_plugins/appcompatcache.py [hunk 1]
```diff
@@ -88,137 +91,37 @@ class AppCompatCachePlugin(interface.WindowsRegistryPlugin):
...
+  _HEADER_SIGNATURES = {
+      # AppCompatCache format signature used in Windows XP.
+      0xdeadbeef: _FORMAT_TYPE_XP,
+      # AppCompatCache format signature used in Windows 2003, Vista and 2008.
+      0xbadc0ffe: _FORMAT_TYPE_2003,
+      # AppCompatCache format signature used in Windows 7 and 2008 R2.
+      0xbadc0fee: _FORMAT_TYPE_7,
+      # AppCompatCache format used in Windows 8.0 and 8.1.
+      0x00000080: _FORMAT_TYPE_8,
+      # AppCompatCache format used in Windows 10
```
Comment (Onager) [inline plaso/parsers/winreg_plugins/appcompatcache.py]: formats, not format (I assume)

Comment (joachimmetz) [reply plaso/parsers/winreg_plugins/appcompatcache.py]: Unclear what this comment is about.

File: plaso/parsers/winreg_plugins/appcompatcache.py [hunk 2]
```diff
@@ -228,34 +131,43 @@ def _CheckSignature(self, value_data):
...
-          self._CACHED_ENTRY_SIGNATURE_8_1]:
+      cached_entry_signature = value_data[signature:signature + 4]
+      if cached_entry_signature == self._CACHED_ENTRY_SIGNATURE_8_1:
         return self._FORMAT_TYPE_10
 
-    return None
+    return format_type
 
-  def _DetermineCacheEntrySize(
+  def _GetCachedEntryDataTypeMap(
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/parsers/winreg_plugins/appcompatcache.py]: ack

File: plaso/parsers/winreg_plugins/appcompatcache.py [hunk 3]
```diff
@@ -264,296 +176,243 @@ def _DetermineCacheEntrySize(
...
-      int: cached entry size or None if not cached..
+      dtfabric.DataTypeMap: data type map which contains a data type definition,
+          such as a structure, that can be mapped onto binary data or None
+          if the data type map is not defined.
 
     Raises:
-      RuntimeError: if the format type is not supported.
+      ParseError: if the cached entry data type map cannot be determined.
     """
     if format_type not in (
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/parsers/winreg_plugins/appcompatcache.py]: Ack

**Comment by joachimmetz** [comment]
Refactored things a bit PTAL

**Pull request merged**